hathor/builder/builder.py

```diff
@@ -565,16 +593,14 @@ def force_memory_index(self) -> 'Builder':
 
     def _get_or_create_wallet(self) -> Optional[BaseWallet]:
         if self._wallet is not None:
-            assert self._wallet_directory is None
-            assert self._wallet_unlock is None
             return self._wallet
 
         if self._wallet_directory is None:
             return None
-        wallet = Wallet(directory=self._wallet_directory)
+        self._wallet = Wallet(directory=self._wallet_directory)
         if self._wallet_unlock is not None:
-            wallet.unlock(self._wallet_unlock)
-        return wallet
```

Comment:
The created wallet was not being persisted here, so I changed it to `self._wallet`. For that, I had to remove the assertions on the top. Adding this comment just to confirm whether there was any reason for the previous code.